geode-core/src/main/java/org/apache/geode/distributed/internal/ClusterDistributionManager.java

```diff
@@ -1082,7 +1082,16 @@ public InternalDistributedMember getId() {
       if (observer != null) {
         observer.beforeSendMessage(this, msg);
       }
-      return sendMessage(msg);
+      Set<InternalDistributedMember> membersNotSent = sendMessage(msg);
+
+      if (membersNotSent != null && !membersNotSent.isEmpty()) {
+        logger.warn(
+            "The following members: {} were not properly sent the message: {}. If unhandled this may"
+                + " cause the result processor to hang while collecting results.",
+            membersNotSent, msg.getShortClassName());
+      }
+
+      return membersNotSent;
```

Comment:
I think the only case where `membersNotSent` will have a value is when `DirectChannel.send()` called from `DistributionImpl.directChannelSend()` generates a `ConnectExceptions` (sic).

In that case `DistributionImpl.directChannelSend()` is already logging warnings `"Failed to send message…"`.

Unless there is some other situation in which `membersNotSent` can contain members, I'd say maybe the message content here, belongs down in `DistributionImpl`.

Now the question is: does it make sense to have a message pertaining to "result processor" down in `ClusterDistributionManager.putOutgoing()` (or even down in `DistributionImpl`). Doing so couples (one of) them to "result processor". What is a "result processor" in this context? Is it a concept is part of the `ClusterDistributionManager` design?

Reply:
Thanks for the observations, Bill! Right before the `"Failed to send message..."` log in `DistributionImpl.directChannelSend()`, we perform the following check:
```
 if (!membership.hasMember(member) || (th instanceof ShunnedMemberException)) {
          continue;
  }
```
which effective skips the log for those cases.  The `ShunnedMemberException` is the one we encountered here.  We could log before checking those cases or log something different within that if block and remove the log we added to `ClusterDistributionManager`.  What do you think is appropriate?

Regarding the "result processor"-specific message, that is a good question.  The "result processor" in this context is `ReplyProcessor21` (so perhaps that should say `reply processor` instead).  I think that's a good call out though, we probably shouldn't assume that if a message is sent there is always a reply processor involved.  It seems like we should remove this log anyway in favor of logging in `DistributionImpl` instead, so maybe we could just stick with the `"Failed to send message..." log there.

Reply:
The log message states - `If unhandled...` what does it then mean to 'handle' the situation?

Reply:
Yeah, I think we will ditch that message in `ClusterDistributionManager` in favor of the one in `DistributionImpl`.  Do you have any opinions on that log message (if we should move the message up before the check or log a different message in that if block)?

Reply:
I've removed the log from `ClusterDistributionManager` and moved the `DistributionImpl` log before the if block.  Let me know if y'all think this is the right approach. @jdeppe-pivotal @Bill

Reply:
...I switched it to just do a debug log in that if block.  Let me know if y'all think that's ok or if we should move the fatal log up (it will also result in failing tests that need to be adjusted due to suspicious string failures).